### PR TITLE
Implemented "attach" command.

### PR DESCRIPTION
**Pull request by MisterOwlPT**
Hello everyone!

I've noticed that the _attach_ command has not yet been implemented and decided to implement it.
With the proposed changes the following is now possible:

```python
from python_on_whales import docker
docker.attach('my_container')  # or docker.container.attach('my_container')

# current shell attaches itself to the container (if existent) shell and 
# all input/output is replicated on all attached shells 
```

Proposed changes also include:
- Support for all CLI arguments;
- Unit tests;
- Documentation update.

### CODE REVIEW

**Comment by gabrieldemarmiesse** [comment]
Hi! I'm currently moving my house so it might take a bit of time before I'm available to work every day on this project like I'm used to. I'll be less busy by the end of next week :)

File: python_on_whales/components/container/cli_wrapper.py [hunk 1]
```diff
@@ -353,9 +370,38 @@ def __init__(self, *args, **kwargs):
...
-    def attach(self):
-        """Not yet implemented"""
-        raise NotImplementedError
+    def attach(
+        self,
+        container: ValidContainer,
+        detach_keys: Optional[str] = None,
+        no_stdin: bool = False,
+        sig_proxy: bool = True,
+    ) -> Union[None, str, Iterable[Tuple[str, bytes]]]:
```
Comment (gabrieldemarmiesse) [inline python_on_whales/components/container/cli_wrapper.py:379]: In which case does it return None and in which case does it returns an iterable for the logs?

If it's possible to return an iterable for the logs, would it be possible to test this in the unit tests without any mocks?

**Review by gabrieldemarmiesse — COMMENTED** [review]
Looks good, just a question

File: python_on_whales/components/container/cli_wrapper.py (hunk 1 above)
Comment (MisterOwlPT) [reply python_on_whales/components/container/cli_wrapper.py]: Hey hi!
Glad to hear from you again :)

Whoops... basically during development I copied existing function headers as a template and did not pay much attention to their return types. Per my understanding,  _'attach' should always return None_ as it only attaches itself to an existing container and does not change/replace it in any way. Different types are instead returned by the 'run' function that launched the container in the first place.

I updated the required functions and their docstrings to reflect this.
No change was made to the existing unit tests.

File: python_on_whales/components/container/cli_wrapper.py [hunk 2]
```diff
@@ -175,6 +175,21 @@ def __repr__(self):
     # --------------------------------------------------------------------
     # public methods
 
+    def attach(
+        self,
+        detach_keys: Optional[str] = None,
+        no_stdin: bool = False,
```
Comment (gabrieldemarmiesse) [inline python_on_whales/components/container/cli_wrapper.py:181]:
```suggestion
        stdin: bool = True,
```

File: python_on_whales/components/container/cli_wrapper.py [hunk 3]
```diff
@@ -175,6 +175,21 @@ def __repr__(self):
...
+        sig_proxy: bool = True,
+    ) -> None:
+        """Attach local standard input, output, and error streams to a running container.
+
+        Alias: `docker.attach(...)`
+
+        See the [`docker.container.attach`](../sub-commands/container.md#attach) command for
+        information about the arguments.
+        """
+        ContainerCLI(self.client_config).attach(self, detach_keys, no_stdin, sig_proxy)
```
Comment (gabrieldemarmiesse) [inline python_on_whales/components/container/cli_wrapper.py:191]:
```suggestion
        ContainerCLI(self.client_config).attach(self, detach_keys, not stdin, sig_proxy)
```

File: python_on_whales/components/container/cli_wrapper.py [hunk 4]
```diff
@@ -353,9 +368,35 @@ def __init__(self, *args, **kwargs):
...
         self.remove = self.remove
 
-    def attach(self):
-        """Not yet implemented"""
-        raise NotImplementedError
+    def attach(
+        self,
+        container: ValidContainer,
+        detach_keys: Optional[str] = None,
+        no_stdin: bool = False,
```
Comment (gabrieldemarmiesse) [inline python_on_whales/components/container/cli_wrapper.py:375]:
```suggestion
        not stdin: bool = True,
```

File: python_on_whales/components/container/cli_wrapper.py [hunk 5]
```diff
@@ -353,9 +368,35 @@ def __init__(self, *args, **kwargs):
...
+            sig_proxy: Proxy all received signals to the process (default true)
+
+        # Raises
+            `python_on_whales.exceptions.NoSuchContainer` if the container does not exists.
+        """
+        self.inspect(container)
+
+        full_cmd = self.docker_cmd + ["attach"]
+        full_cmd.add_simple_arg("--detach-keys", detach_keys)
+        full_cmd.add_flag("--no-stdin", no_stdin)
```
Comment (gabrieldemarmiesse) [inline python_on_whales/components/container/cli_wrapper.py:395]:
```suggestion
        full_cmd.add_flag("--no-stdin", not stdin)
```

File: tests/python_on_whales/components/test_container.py [hunk 6]
```diff
@@ -631,3 +633,72 @@ def test_prune():
...
+    )
+
+
+@patch("python_on_whales.components.container.cli_wrapper.ContainerCLI.inspect")
+@patch("python_on_whales.components.container.cli_wrapper.run")
+def test_attach_no_stdin_argument(run_mock: Mock, inspect_mock: Mock) -> None:
+
+    test_container_name = "test_dummy_container"
+
+    docker.attach(test_container_name, no_stdin=True)
```
Comment (gabrieldemarmiesse) [inline tests/python_on_whales/components/test_container.py:683]:
```suggestion
    docker.attach(test_container_name, stdin=False)
```

File: python_on_whales/components/container/cli_wrapper.py [hunk 7]
```diff
@@ -353,9 +368,35 @@ def __init__(self, *args, **kwargs):
...
+        sig_proxy: bool = True,
+    ) -> None:
+        """Attach local standard input, output, and error streams to a running container
+
+        Alias: `docker.attach(...)`
+
+        # Arguments
+            container: The running container to attach to
+            detach_keys: Override the key sequence for detaching a container
+            no_stdin: Do not attach STDIN
```
Comment (gabrieldemarmiesse) [inline python_on_whales/components/container/cli_wrapper.py:385]:
```suggestion
            stdin: Attach STDIN
```

**Review by gabrieldemarmiesse — CHANGES_REQUESTED** [review]

File: python_on_whales/components/container/cli_wrapper.py [hunk 8]
```diff
@@ -353,9 +368,35 @@ def __init__(self, *args, **kwargs):
...
         self.remove = self.remove
 
-    def attach(self):
-        """Not yet implemented"""
-        raise NotImplementedError
+    def attach(
+        self,
+        container: ValidContainer,
+        detach_keys: Optional[str] = None,
+        not stdin: bool = True,
```
Comment (gabrieldemarmiesse) [inline python_on_whales/components/container/cli_wrapper.py:375]:
```suggestion
        stdin: bool = True,
```

**Review by gabrieldemarmiesse — CHANGES_REQUESTED** [review]

**Pull request merged**